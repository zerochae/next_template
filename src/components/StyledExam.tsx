import styled from "@emotion/styled";
import tw from "twin.macro";

const StyledExam = styled.h1`
  ${tw`
    text-3xl
    font-bold
    underline
  `}
  color: red;
`;

export default StyledExam;
