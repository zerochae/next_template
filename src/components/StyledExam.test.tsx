import { render, screen } from "@testing-library/react";
import { matchers } from "@emotion/jest";
import StyledExam from "./StyledExam";

expect.extend(matchers);

describe("style test suite", () => {
  test("styled test case", () => {
    render(<StyledExam>헤딩</StyledExam>);
    const headingEl = screen.getByRole("heading", { level: 1 });
    expect(headingEl).toBeInTheDocument();
  });

  test("snapshot", () => {
    render(<StyledExam>헤딩</StyledExam>);
    const headingEl = screen.getByRole("heading", { level: 1 });
    expect(headingEl).toMatchSnapshot();
  });

  test("style rule", () => {
    render(<StyledExam>헤딩</StyledExam>);
    const headingEl = screen.getByRole("heading", { level: 1 });
    expect(headingEl).toHaveStyleRule("color", "red");
  });

  test("to have style", () => {
    render(<StyledExam>헤딩</StyledExam>);
    const headingEl = screen.getByRole("heading", { level: 1 });
    expect(headingEl).toHaveStyle("color:red");
  });
});
