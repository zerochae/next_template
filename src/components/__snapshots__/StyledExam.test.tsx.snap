// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`style test suite snapshot 1`] = `
<h1
  class="css-1j84qhj-StyledExam e1d9arlk0"
>
  헤딩
</h1>
`;
